wemake_python_styleguide/visitors/ast/naming.py

```diff
@@ -126,6 +128,38 @@ def visit_Name(self, node: ast.Name) -> None:
 
         self.generic_visit(node)
 
+    # FIXME: Remove "noqa" after fixing problem with comprehension
+    # variables being counted as locals.
+    def visit_ClassDef(self, node: ast.ClassDef) -> None:  # noqa: Z210
+        """
+        Used to find upper attribute declarations.
+
+        Raises:
+            WrongVariableNameViolation
+            TooShortVariableNameViolation
+            PrivateNameViolation
+
+        """
+        top_level_assigns = [
+            sub_node
+            for sub_node in node.body
+            if isinstance(sub_node, ast.Assign)
+        ]
+
+        upper_case_violations = [
+            target
```

Comment:
Can you please open a new issue for `counting comprehensions as locals`?

Reply:
I'm going to, didn't have time when I discovered it. :)

Reply:
Here it is: https://github.com/wemake-services/wemake-python-styleguide/issues/247